FunctionalEquivalence/FunctionalEquivalence.wdl

```diff
@@ -0,0 +1,508 @@
+version 1.0
+
+import "../BenchmarkVCFs/BenchmarkVCFs.wdl" as BenchmarkVCFs
+
+workflow FunctionalEquivalence {
+    input{
+        Array[String] sample_id
+        Array[Boolean] include_in_fe_analysis
+        Array[Boolean] create_roc_plot
+        Array[String] dataset
+        Array[String] replicate_no
+
+        Array[String] confidence_intervals
+
+        Array[String] tool1_vcf
+        Array[String] tool1_vcf_index
+        Array[String] tool2_vcf
+        Array[String] tool2_vcf_index
+
+        Array[String] truth_vcf
+        Array[String] truth_vcf_index
+
+        Array[File]? stratIntervals
+        Array[String]? stratLabels
+
+        File reference
+        String referenceVersion = "hg38"
+        File refIndex
+        File refDict
+        File hapMap
+
+        String tool1_label
+        String tool2_label
+        String? additional_label
+
+        Boolean signed_difference = false
+
+        String f1_script_http_url = "https://github.com/broadinstitute/palantir-workflows/raw/main/FunctionalEquivalence/scripts/f1_evaluation.py"
+        String fe_script_http_url = "https://github.com/broadinstitute/palantir-workflows/raw/main/FunctionalEquivalence/scripts/fe_evaluation.py"
+        String roc_script_http_url = "https://github.com/broadinstitute/palantir-workflows/raw/main/FunctionalEquivalence/scripts/plot_rocs.R"
+
+        String gatkTag="4.0.11.0"
+        Boolean passingOnly=true
+        Boolean requireMatchingGenotypes=true
+        String vcfScoreField = "QUAL"
+        Int? threadsVcfEval=2
+        Int? preemptible = 3
+    }
+        
+    scatter (i in range(length(sample_id))) {
+        if (truth_vcf[i] != "null" && (include_in_fe_analysis[i] || create_roc_plot[i])) {
```

Comment:
This doesn't work unfortunately, because if a table column has entries ["a", null, "b"], the resulting Array[String?] would be ["a", "b"], and then the mapping between the samples and their truth VCFs gets lost. So this is the only solution I could think of. Happy to change if anyone has a better idea

Reply:
oh man, that is super annoying